framework/test/unit/node/consensus/consensus.spec.ts

```diff
@@ -506,21 +514,32 @@ describe('consensus', () => {
 	describe('execute', () => {
 		let block: Block;
 		let apiContext: APIContext;
+		let verifyStateRootMock: jest.Mock;
 
-		beforeEach(async () => {
-			block = await createValidDefaultBlock({ header: { height: 2 } });
-			apiContext = createTransientAPIContext({});
-			jest.spyOn(chain, 'saveBlock').mockResolvedValue();
-			jest.spyOn(consensus, 'finalizedHeight').mockReturnValue(0);
-
-			jest.spyOn(stateMachine, 'verifyBlock').mockResolvedValue();
-			jest.spyOn(stateMachine, 'executeBlock').mockResolvedValue();
-			jest.spyOn(consensus.events, 'emit');
+		describe('block verification', () => {
+			beforeEach(async () => {
+				block = await createValidDefaultBlock({ header: { height: 2 } });
+				apiContext = createTransientAPIContext({});
+				jest.spyOn(chain, 'saveBlock').mockResolvedValue();
+				jest.spyOn(consensus, 'finalizedHeight').mockReturnValue(0);
+				jest.spyOn(stateMachine, 'verifyBlock').mockResolvedValue();
+				jest.spyOn(stateMachine, 'executeBlock').mockResolvedValue();
+				jest.spyOn(bftAPI, 'getBFTParameters').mockResolvedValue({
+					validatorsHash: block.header.validatorsHash,
+				} as never);
+				jest.spyOn(consensus.events, 'emit');
+				consensus['_db'] = dbMock;
+				verifyStateRootMock = jest.fn();
+				consensus['_verifyStateRoot'] = verifyStateRootMock;
+				verifyStateRootMock.mockReturnValue(undefined);
```

Comment:
Is there a reason why you didn't use something like `jest.spyOn().mockImplementation()` here so we can reset the mock when we need in a test case?

Reply:
jest.spyOn() doesn't work for private function

Reply:
I think I made it work like `jest.spyOn(class, '_privateMethod' as any).mockImplementation()` but it's open to a debate whether it's a good practice 😄 Got it.

Reply:
ah nice, I mean if it works then lets use it from next time